test/unit/test-polyfill-custom-elements.js

```diff
@@ -14,7 +14,78 @@
  * limitations under the License.
  */
 
-import {copyProperties} from '../../src/polyfills/custom-elements';
+import {copyProperties, install} from '../../src/polyfills/custom-elements';
+
+describes.realWin(
+  'install patches',
+  {skipCustomElementsPolyfill: true},
+  (env) => {
+    let win;
+    let innerHTMLProto;
+    beforeEach(function () {
+      win = env.win;
+
+      innerHTMLProto = win.Element.prototype;
+      if (!Object.getOwnPropertyDescriptor(innerHTMLProto, 'innerHTML')) {
+        innerHTMLProto = win.HTMLElement.prototype;
+        if (!Object.getOwnPropertyDescriptor(innerHTMLProto, 'innerHTML')) {
+          return this.skip();
+        }
+      }
+
+      // We want to test the full polyfill.
+      delete win.Reflect;
+      delete win.customElements;
+    });
+
+    it('handles non-configurable innerHTML accessor', () => {
+      // Use strict is important, as strict mode code will throw an error when
+      // trying to redefine a non-configurable property.
+      'use strict';
+
+      Object.defineProperty(innerHTMLProto, 'innerHTML', {configurable: false});
+      install(win, function () {});
+
+      class Test extends win.HTMLElement {}
+
+      expect(() => {
+        win.customElements.define('x-test', Test);
+      }).not.to.throw();
+    });
+
+    it('handles missing innerHTML descriptor', () => {
```

Comment:
Can you add a comment to each case about which browser is affected? Safari 9 and below, Yandex, IE11.

Reply:
Done.